component/src/main/java/io/siddhi/extension/execution/file/FileArchiveExtension.java

```diff
@@ -0,0 +1,422 @@
+            }
+            if (attributeExpressionExecutors.length == 5) {
+                if (attributeExpressionExecutors[4] == null) {
+                    throw new SiddhiAppValidationException("Invalid input given to regex " +
+                            "(third argument) file:archive() function. Argument cannot be null");
+                }
+                Attribute.Type thirdAttributeType = attributeExpressionExecutors[4].getReturnType();
+                if (!(thirdAttributeType == BOOL)) {
+                    throw new SiddhiAppValidationException("Invalid parameter type found for regex " +
+                            "(third argument) of file:archive() function, required " + Attribute.Type.STRING +
+                            " but found " + firstAttributeType.toString());
+                }
+            }
+        } else {
+            throw new SiddhiAppValidationException("Invalid no of arguments passed to file:archive() function, "
+                    + "required 2 or 3, but found " + attributeExpressionExecutors.length);
+        }
+        return null;
+    }
+
+    @Override
+    public List<Attribute> getReturnAttributes() {
+        List<Attribute> attributes = new ArrayList<>();
+        attributes.add(new Attribute("isSuccess", BOOL));
+        return attributes;
+    }
+
+    @Override
+    public ProcessingMode getProcessingMode() {
+        return ProcessingMode.BATCH;
+    }
+
+    @Override
+    public void start() {
+
+    }
+
+    @Override
+    public void stop() {
+
+    }
+
+    /**
+     * Zip it
+     *
+     * @param zipFile output ZIP file location
+     */
+    private void zip(String sourceFileUri, String zipFile, List<String> fileList) {
+        byte[] buffer = new byte[1024];
+        FileInputStream in = null;
+        String filePath = null;
+        ZipOutputStream zos = null;
+        try {
+            if (!zipFile.endsWith(TAR_FILE_EXTENSION)) {
+                zipFile = zipFile.concat("." + ZIP_FILE_EXTENSION);
+            }
+            FileOutputStream fos = new FileOutputStream(zipFile);
+            zos = new ZipOutputStream(fos);
+            if (log.isDebugEnabled()) {
+                log.debug("Output to Zip : " + zipFile + " started for folder/ file: " + sourceFileUri);
+            }
+            for (String file : fileList) {
+                if (log.isDebugEnabled()) {
+                    log.debug("File Adding : " + file + " to " + zipFile + ".");
+                }
+                ZipEntry ze = new ZipEntry(file);
+                zos.putNextEntry(ze);
+                filePath = sourceFileUri + File.separator + file;
+                in = new FileInputStream(sourceFileUri + File.separator + file);
+                int len;
+                while ((len = in.read(buffer)) > 0) {
+                    zos.write(buffer, 0, len);
+                }
+                in.close();
+            }
+            zos.closeEntry();
+            //remember close it
+            zos.close();
```

Comment:
isn't `zos` also getting closed in finally?

Reply:
@grainier fixed in https://github.com/siddhi-io/siddhi-io-file/pull/72/commits/c2d3273f60ae4d82310996901fd59f38bb4c7975